src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouped: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyOfItem: string = String(item[key]);
```

Comment:
You don`t have to specify string type here
https://www.typescripttutorial.net/typescript-tutorial/typescript-type-inference/